lib/json_api_client/associations/has_one.rb

```diff
@@ -5,7 +5,12 @@ class Association < BaseAssociation
         def from_result_set(result_set)
           result_set.first
         end
+
+        def load_records(data)
+          record_class = Utils.compute_type(klass, data["type"].classify)
```

Comment:
After testing I have found there is a bug here, when the payload is in dash case running classify on the `type` we get unexpected results. `"skill-level". classify` will return `Skill-level`. So what we need to do here is `data["type"].underscore. classify`

Reply:
If fixed this locally with
```
  JsonApiClient::Associations::HasOne::Association.class_eval do
    def load_records(data)
      record_class = JsonApiClient::Utils.compute_type(klass, data["type"].underscore.classify)
      record_class.load id: data["id"]
    end
  end
```